packages/logging_cloudwatch/aws_logging_cloudwatch/test/cloudwatch_logger_plugin_test.dart

```diff
@@ -132,25 +131,163 @@ void main() {
         (_) async => Future<PutLogEventsResponse>.value(PutLogEventsResponse()),
       );
 
+      when(() => mockCloudWatchLogsClient.putLogEvents(any())).thenAnswer(
+        (_) => mockPutLogEventsOperation,
+      );
+
       when(() => mockCloudWatchLogStreamProvider.logStream)
           .thenAnswer((_) async => Future<String>.value('log stream name'));
 
+      when(() => mockQueuedItemStore.addItem(any(), any()))
```

Comment:
Is there any benefit to mocking over using the in-mem store? I would generally mock a single layer per test, in this case just the CloudWatch SDK.

Reply:
mocking dependencies make the tests faster, reliable and can isolate the code under test from its dependencies.

Reply:
Then there should be tests for the log stream provider as well since that's currently not being tested.

Reply:
agree, will add them in a seperate PR.